ide-support/META-INF/MANIFEST.MF

```diff
@@ -7,7 +7,6 @@ Bundle-SymbolicName: org.ec4j.core.services
 Bundle-Version: 0.1.0.qualifier
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
 Bundle-ActivationPolicy: lazy
-Export-Package: org.ec4j.core.services,
- org.ec4j.core.services.completion,
- org.ec4j.core.services.validation
+Export-Package: org.ec4j.core.ide,
+ org.ec4j.core.ide.completion
```

Comment:
Package name should be not just "org.ec4j.ide", why core? If you refactor package name, please refactor "Bundle-SymbolicName" too like 

> Bundle-SymbolicName: org.ec4j.ide

Reply:
> Package name should be not just "org.ec4j.ide", why core?

It is a common convention that that artifacts living under the same Maven groupId use the same common package name prefix. In other words, because both `ec4j-ide-support` and `ec4j-core` live under groupId `org.ec4j.core`, their package names should have a common prefix specific to that `groupId`. 

This is e.g. practical when working on the source code level and you see an `org.acme.foo.bar.Baz` class somewhere and you wonder which artifact is it coming from, you can conclude quite quickly that the groupId is some prefix of the package. You can assume it is either `org.acme.foo.bar` or `org.acme.foo` and you can exclude all other options, esp. `org.acme.bar`.

> If you refactor package name, please refactor "Bundle-SymbolicName" too like

Good catch, will do.

Reply:
> If you refactor package name, please refactor "Bundle-SymbolicName" too like

Done in e8e927d

Reply:
I tell me why we have org.ec4j.core package? Perhaps we should have just org.ec4j ? When I started the project I have used "*.core "and "*.services", today whe have "*.core" and "*.core.ide", I find in this case "core" is not very helpful, no?

Reply:
> I tell me why we have org.ec4j.core package? Perhaps we should have just org.ec4j ?

As I tried to explain once already, in this way, we keep us the option open to add further git repos (= groupIds ) under the `org.ec4j` organization in such a way that naming clashes will be impossible.

What you propose is technically possible. We could remove the `core` segment from groupIds, artifactIds, from Java package names and from OSGi Bundle-SymbolicNames. Such state would be perfectly consistent and concise. 

So far so good, but once we decide to add a new project (= new groupId, = new Java package prefix ) under the `org.ec4j` organization, we will have only naming prefixes at our disposal that can clash with the existing `org.ec4j` prefix. Say that we choose the `org.ec4j.foo` prefix for the new project (groupId `org.ec4j.foo`, all its Java package names will start with `org.ec4j.foo` and OSGi  Bundle-SymbolicNames will also start with `org.ec4j.foo`). We would have to take special care in the exiting `org.ec4j` project to avoid using `org.ec4j.foo` java package and `org.ec4j.foo` OSGi bundle name, because otherwise, we'd have a naming clash. Such a clash would be impossible if the old project used `org.ec4j.core` since the very beginning. 

I admit I may be deformed by being used to work with many independent orgs, repos and artifacts and I admit that the advantages of what I propose may show up only when the number of projects exceeds certain level. But at that point, it is usually too complicated to change the naming.

There are actually one more reason for using fully separate branches of the package tree for projects and (and actually also for artifacts):

Logging, tracing, profiling: all these techniques typically work on the level of a Java package subtree. Say, you'd want to log/profile/trace in a greater detail for just classes coming from the core artifact, whose Java package name `org.ec4j` happens to be a prefix of many classes from other artifacts of your organization you actually do not care for, how would you do that? I can tell you from my experience, that it sucks. One typically spends a lot of time with searching what to exclude so that one gets only what one wants to see. Logging/tracing/profiling is so much easier when artifacts use fully independent branches of the package tree

Reply:
Ok I understand more now. Thans for the explanation!